### PR TITLE
Fix portal surround node and add guid

### PR DESCRIPTION
**Pull request by JohnStuartBarrie**
Fix weird behaviour selecting activity options from contact activity - hover over an option displays the other options under the hovered option. Fixed by using `<span>`. Also adding `guid` to `Portal` enter and exit divs to aid automation testing.

### TODO

- [x] Add `guid` to Portal placeholder `<span>` and Portal content.
- [x] Update release notes

### Screenshot
screenshot 2018-03-05 17 03 47

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# 3.1.1
+
+## Improvements
+
+A unique id has been added to the dropdown component and it's Portal list.
```
Comment (andrewjtait) [inline CHANGELOG.md]: the unique id should be added to the portal, not the dropdown

File: src/components/dropdown/__spec__.js [hunk 2]
```diff
@@ -926,16 +926,31 @@ describe('Dropdown', () => {
...
+      
+      it('renders the carbon-dropdown__list on focus ', () => {
+        wrapper.find('.carbon-dropdown__input').simulate('focus');
+        const portalDropdownList = wrapper.find(Portal).find('.carbon-dropdown__list');
+        expect(portalDropdownList).toMatchSnapshot();
+      });
+
+      it('renders the correct data-portal-entrance id', () => {
+        wrapper.find('.carbon-dropdown__input').simulate('focus');
+        expect(wrapper.find('[data-portal-entrance="'+wrapper.instance().uniqueid+'"]').length).toBe(1);
```
Comment (toni-leigh) [inline src/components/dropdown/__spec__.js]: Won't interpolation work: 

```
`[data-portal-entrance=${wrapper.instance().uniqueid}"]`
```

File: src/components/dropdown/__spec__.js [hunk 3]
```diff
@@ -926,16 +926,31 @@ describe('Dropdown', () => {
...
+      });
 
-      wrapper.find('.carbon-dropdown__input').simulate('focus');
-      const portalDropdownList = wrapper.find(Portal).find('.carbon-dropdown__list');
-      expect(portalDropdownList).toMatchSnapshot();
+      it('renders the correct data-portal-exit id', () => {
+        wrapper.find('.carbon-dropdown__input').simulate('focus');
+        expect(wrapper.find(Portal).find('[data-portal-exit="'+wrapper.instance().uniqueid+'"]').length).toBe(1);
```
Comment (toni-leigh) [inline src/components/dropdown/__spec__.js]: Also, interpolation?

File: src/components/dropdown/dropdown.js [hunk 4]
```diff
@@ -143,6 +143,8 @@ const Dropdown = Input(InputIcon(InputLabel(InputValidation(class Dropdown exten
      */
     this.visibleValue = '';
 
+    this.visibleValue = '';
```
Comment (ianoxley) [inline src/components/dropdown/dropdown.js]: This looks like it has been duplicated?

File: src/components/dropdown/__spec__.js [hunk 5]
```diff
@@ -925,18 +925,6 @@ describe('Dropdown', () => {
...
-      const wrapper = mount(
-        <Dropdown
-          name='foo'
-          options={ Immutable.fromJS([{ id: 1, name: 'foo' }, { id: 2, name: 'bar' }]) } value='1'
-        />);
-
-      wrapper.find('.carbon-dropdown__input').simulate('focus');
-      const portalDropdownList = wrapper.find(Portal).find('.carbon-dropdown__list');
-      expect(portalDropdownList).toMatchSnapshot();
-    });
```
Comment (ianoxley) [inline src/components/dropdown/__spec__.js]: Similarly, I'm not sure why this spec has been removed?

File: src/components/dropdown/__snapshots__/__spec__.js.snap [hunk 6]
```diff
@@ -1,29 +0,0 @@
...
-    data-element="option"
-    key="bar2"
-    onClick={[Function]}
-    onMouseOver={[Function]}
-    value={2}
-  >
-    bar
-  </li>
-</ul>
-`;
```
Comment (ianoxley) [inline src/components/dropdown/__snapshots__/__spec__.js.snap]: Just wondering why this snapshot has been removed?

File: src/components/dropdown/__spec__.js (hunk 5 above)
Comment (whomwah) [reply src/components/dropdown/__spec__.js]: Yes, that's my mistake. When I was removing the Portal `guid` code and specs I removed too much. I will re-add. That's why the snapshot is missing too.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,9 @@
+# 3.1.1
+
+## Improvements
+
+A unique id has been added to the Portal component entrance and exit nodes.
```
Comment (ianoxley) [inline CHANGELOG.md]: It'd be better if this explained why this is an improvement, or what problem adding the unique id fixes.

**Pull request merged**